### ISSUE TITLE
Allow for enabling termination protection

### ISSUE THREAD

**Issue by kddejong** [cloudtools/stacker#669]
Not sure if protect should do both but it would be nice to have Termination Protection enabled on stacks created and the logic around it when doing a deletion.  While Protect kind of works it doesn't protect us for folks outside of stacker

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-protect-stacks.html

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation.html#CloudFormation.Client.create_stack

**Comment by kddejong** [comment]
exactly.  I'm cool with the flag being removed (turned off) during a destroy action.  Or at least that is what I was going to do with the hooks you provide.  So I'm partial to that strategy.  

This would be to help prevent anyone accidentally deleting the stack from the console/cli or outside of our stacker automation pipelines.

Thanks.

**Comment by phobologic** [comment]
Yeah, that sounds good.  Probably wouldn't be too hard to deal with either.  Could easily see this being the default mode as well.
